### PR TITLE
QtWebEngine: Port QtWebEngine to Haiku

### LINKED ISSUE #2512: Add recipe for QtWebEngine
The Qt WebEngine add-on module provides classes and functions for embedding web content in applications and providing Chromium's fast moving web capabilities into Qt.

URL: https://wiki.qt.io/QtWebEngine
Docs: http://doc.qt.io/qt-5/qtwebengine-index.html

Closes: https://github.com/haikuports/haikuports/issues/3638

### PR DESCRIPTION
**Pull request by kenmays**
This is the PR to port QtWebEngine to Haiku. Currently, Qt 5.12.3 is used for Haiku. 
Demo:: https://youtu.be/elqmjspONk8

NOTE: **W-I-P, Do not merge. Help wanted** 
- 5/14/2019 - KapiX has tested and provided his patches for QtWebEngine 5.11.1. I migrated his patches to v5.12.3 We had some crash issues during the earlier QtWebEngine 5.11.x testing. Reviewing.
- 5/22/2019 - Successfully merged 5.11.x patches to 5.12.3. Bumped to hrev53158 for build test. 

Closes #2512
Relates to:  #3638

### CODE REVIEW

**Comment by extrowerk** [comment]
I think this is too early and will be superseded, so just extra troubles... But whatever.

**Comment by KapiX** [comment]
FYI I'm currently migrating 5.11.1 patchset to 5.12.3 but a lot of stuff has changed, so it will take some time.

**Comment by kenmays** [comment]
KapiX -Cleaned up the patchset against QtWebEngine 5.12.3. Posted for your review.

**Comment by extrowerk** [comment]
@kenmays: does it builds?

**Comment by kenmays** [comment]
Just bumped to hrev53112... maybe this weekend.

**Comment by KapiX** [comment]
I didn't try but I can say just by looking at it that it doesn't. It is 5.11.1 patchset and it doesn't apply cleanly (this I tried). This PR as is cannot be merged.

**Comment by kenmays** [comment]
@Kapix - W-I-P. The 5.12.3 patchset is the result of what merged to my 5.12.3 tree - from your 5.11.1 patchset. Still needs some work.

File: dev-qt/qtwebengine/patches/qtwebengine-5.12.3.patchset [hunk 1]
```diff
@@ -0,0 +1,4832 @@
+Date: Sun Sept 9 2018 08:10:16 +0000 
+From: Ken Mays <kmays2000@gmail.com>
+Subject: Qt WebEngine 5.11.2 patchset
+
+From: Jan Steffens <heftig@archlinux.org>
+Date: 2017-09-19 23:30:08 -0700
+Subject: This is required for QtWebEngine to show the patent-free LCD rendering. Without this patch, only grayscale rendering is used.
```
Comment (extrowerk) [inline dev-qt/qtwebengine/patches/qtwebengine-5.12.3.patchset]: Is this relevant?

Comment (kenmays) [reply dev-qt/qtwebengine/patches/qtwebengine-5.12.3.patchset]: Still applies to QTWebEngine 5.12.3. Haiku/AGG - I'd leave it for now.

Comment (extrowerk) [reply dev-qt/qtwebengine/patches/qtwebengine-5.12.3.patchset]: I already asked you more than once in your earlier qtwebengine PRs why is this required, but  so fa got no explanationr, so i assume this is just something what you want to have, which is nice, but this doesn't makes the Haiku port easier (actually it **can** even create troubles), and technically this have nothing to do with the Haiku port, so i still say **currently** it is unnecessary. If it builds fine, then can we start to experiment with 3rdparty patches, and enable more features.

**Comment by kenmays** [comment]
@KapiX , @extrowerk - I'm going to offline this. Will work on a final implementation in the meantime. Thanks for your previous help.

**Pull request closed without merging**